bip32/src/defaults.rs

```diff
@@ -1,12 +1,16 @@
 use crate::enc::XKeyEncoder;
 
-/// The default encoder, selected by feature flag
-#[cfg(feature = "mainnet")]
-pub type Encoder = crate::enc::MainnetEncoder;
-
-/// The default encoder, selected by feature flag
-#[cfg(feature = "testnet")]
-pub type Encoder = crate::enc::TestnetEncoder;
+cfg_if::cfg_if! {
+    if #[cfg(feature = "mainnet")] {
+        /// The default encoder, selected by feature flag
+        pub type Encoder = crate::enc::MainnetEncoder;
+    } else if #[cfg(feature = "testnet")] {
```

Comment:
this is a behavior change, no? it causes `mainnet` to override `testnet` instead of causing a compilation error if both are active?

Reply:
You're right, this will change behavior. Maybe just add a compile error to prompt?